core/src/main/java/com/epam/esm/model/CustomSearchRequest.java

```diff
@@ -0,0 +1,20 @@
+package com.epam.esm.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.io.Serializable;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class CustomSearchRequest implements Serializable {
+    private String namePrefix;
+    private String descriptionPrefix;
+    private String tagNamePrefix;
```

Comment:
based on requirements it is expected tag name not just prefix

Reply:
by this field we can also find tag by full name